examples/generic_watcher.rs

```diff
@@ -0,0 +1,49 @@
+// This example shows how to use kube with dynamically known resource kinds.
+
+use color_eyre::Result;
+use futures::prelude::*;
+use kube::{
+    api::{DynamicObject, ErasedKind, ListParams, Meta, StringRef},
+    Api, Client,
+};
+use kube_runtime::{utils::try_flatten_applied, watcher};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    env_logger::init();
+
+    // alternatively, you can:
+    // 1) take them CLI arguments
+    // 2) use dynamic discovery apis on kube::Client and e.g. watch all
+    // resources in cluster.
+    let group = std::env::var("GROUP").expect("GROUP not set");
+    let group = group.trim();
+    let version = std::env::var("VERSION").expect("VERSION not set");
+    let version = version.trim();
+    let api_version = if group.is_empty() {
+        version.to_string()
+    } else {
+        format!("{}/{}", group, version)
+    };
+    let kind = std::env::var("KIND").expect("KIND not set");
+    let kind = kind.trim();
+
+    let erased_kind = ErasedKind {
+        group: StringRef::Dynamic(group.into()),
+        version: StringRef::Dynamic(version.into()),
+        api_version: StringRef::Dynamic(api_version.into()),
+        kind: StringRef::Dynamic(kind.into()),
+    };
+
+    let client = Client::try_default().await?;
+    let api = Api::<DynamicObject>::all_with(client, &erased_kind);
```

Comment:
Not sure if this is achievable, but ideally, I feel like this could look like:
```rust
let gvk = GroupVersionKind::group(group).version(version).kind(kind);
let api: Api<DynamicObject> = gvk.into_api(client);
```

which looks very similar to the current `DynamicResource`, but don't worry about breaking `DynamicResource`; it already doesn't work.